superchain/configs/mainnet/superchain.toml

```diff
@@ -12,3 +12,23 @@ granite_time = 1726070401 # Tue 11 Sep 2024 16:00:01 UTC
   chain_id = 1
   public_rpc = "https://ethereum-rpc.publicnode.com"
   explorer = "https://etherscan.io"
+
+[l1.ImplementationAddresses."op-contracts/v1.3.0"]
+  AddressManager = "0xdE1FCfB0851916CA5101820A69b13a4E276bd81F"
+  L1CrossDomainMessenger = "0xD3494713A5cfaD3F5359379DfA074E2Ac8C6Fd65"
+  L1ERC721Bridge = "0xAE2AF01232a6c4a4d3012C5eC5b1b35059caF10d"
+  L1StandardBridge = "0x64B5a5Ed26DCb17370Ff4d33a8D503f0fbD06CfF"
+  L2OutputOracle = "0xF243BEd163251380e78068d317ae10f26042B292"
+  OptimismMintableERC20Factory = "0xE01efbeb1089D1d1dB9c6c8b135C934C0734c846"
+  OptimismPortal = "0x2D778797049FE9259d947D1ED8e5442226dFB589"
+  SystemConfig = "0xba2492e52F45651B60B8B38d4Ea5E2390C64Ffb1"
```

Comment:
My one hesitation with this change is it breaks the current pattern of storing "standard data" in the `validation/standard` module+package. 

There are a few options:
1. Live with a mixture / scatter of standard data locations
2. Move this into `validation/standard`
3. Resolve in the other direction, and put standard data into the `superchain` module directly.

We have been so far working under the assumption that `superchain` is probably the right location for data which will be consumed by downstream applications, and `validation` is not really designed to be consumed. 

@mds1 @vdamle @bitwiseguy curious for your ideas on this.

Reply:
> What do you think about having this data live in validation/standard for consistency, and superchain.toml just specifies the latest standard release, such as:

@mds1 @geoknee @bitwiseguy Thank you for the comments and the proposal for what the spelling should look like and where it should live.

I'm in agreement that moving the "standard data" to the `validation/standard` makes sense since that is where most data is sourced/validated from. With that in mind, I have the following format in mind for storing the `{ "version", "implementation_address" }` in `validation/standard/standard-versions.toml`:

```
# Fault Proofs https://github.com/ethereum-optimism/optimism/releases/tag/op-contracts%2Fv1.4.0
["op-contracts/v1.4.0"]
address_manager = { version = "", implementation_address = ""}
l1_cross_domain_messenger = { version = "2.3.0", implementation_address = ""}
l1_erc721_bridge = { version = "2.1.0", implementation_address = ""}
l1_standard_bridge = { version = "2.1.0", implementation_address = ""}
optimism_mintable_erc20_factory = { version = "1.9.0", implementation_address = ""}
optimism_portal = { version = "3.10.0", implementation_address = ""}
system_config = { version = "2.2.0", implementation_address = ""}
protocol_versions = { version = "1.0.0", implementation_address = ""}
anchor_state_registry = { version = "1.0.0", implementation_address = ""}
delayed_weth = { version = "1.0.0", implementation_address = ""}
dispute_game_factory = { version = "1.0.0", implementation_address = ""}
fault_dispute_game = { version = "1.2.0", implementation_address = ""}
permissioned_dispute_game = { version = "1.2.0", implementation_address = ""}

# Fault Proofs https://github.com/ethereum-optimism/optimism/releases/tag/op-contracts%2Fv1.4.0
["op-contracts/v1.3.0"]
address_manager = { version = "", implementation_address = ""}
l1_cross_domain_messenger = { version = "2.3.0", implementation_address = ""}
l1_erc721_bridge = { version = "2.1.0", implementation_address = ""}
l1_standard_bridge = { version = "2.1.0", implementation_address = ""}
optimism_mintable_erc20_factory = { version = "1.9.0", implementation_address = ""}
optimism_portal = { version = "3.10.0", implementation_address = ""}
system_config = { version = "2.2.0", implementation_address = ""}
protocol_versions = { version = "1.0.0", implementation_address = ""}
anchor_state_registry = { version = "1.0.0", implementation_address = ""}
delayed_weth = { version = "1.0.0", implementation_address = ""}
dispute_game_factory = { version = "1.0.0", implementation_address = ""}
fault_dispute_game = { version = "1.2.0", implementation_address = ""}
permissioned_dispute_game = { version = "1.2.0", implementation_address = ""}
```

If we do decide on this, one aspect that is somewhat odd to me is that the table of addresses for each network is in the "config" section of the superchain: https://github.com/ethereum-optimism/superchain-registry/blob/main/superchain/configs/mainnet/base.toml#L38-L54 . Perhaps this also ought to live in the `validation` directory? Not suggesting I want to tackle that refactoring as well in this PR, but wanted to solicit feedback while we are on this topic!

Reply:
I'm in favour of moving all of the information into `validation/standard`. I do like the idea of separating out the declaration of which `op-contracts` tag is _currently_ standard (e.g. latest governance approved versions) from the list of historical/current/future tags. For the latter list I like having each tag be a table, with a row for each contract. The value can definitely be a `version`, `implementation_address` pair.

@vdamle your format looks like a mix of toml and json -- what would it look like as pure toml?

Reply:
@geoknee No, this is pure TOML - [inline tables](https://toml.io/en/v1.0.0#inline-table)

Also supported by `burntsushi/toml`: https://github.com/BurntSushi/toml/blob/master/_example/example.toml#L26-L31

Reply:
I'm also in favor of moving all to this information into the `validation/standard` dir. 

I like the data format of `{ version, implementation_address }` for each contract, but I'm curious what `implementation_address` means here. Is it used because all of these contracts are proxies and we actually want the underlying logic contract address? Or could we just use `address` instead of `implementation_address`?

> the table of addresses for each network is in the "config" section of the superchain
Each chain's config file contains information that is specific to that chain. The `[addresses]` field contains addresses of contracts and roles for that chain, which vary from chain-to-chain. Some of the addresses are used by downstream software that consumes the configs (e.g. by op-node [here](https://github.com/ethereum-optimism/optimism/blob/develop/op-node/rollup/superchain.go#L87-L88)) so I think that info should probably remain in the configs.

Reply:
> Is it used because all of these contracts are proxies and we actually want the underlying logic contract address? Or could we just use address instead of implementation_address?

They are all not proxies. For the ones that are, the `implementation_address` is for the implementation contract behind the proxy. The concern I have with naming the field as `address` could make it confusing to an end user. As your question would imply, for the contracts that are not proxies, we'll leave the implementation address as an empty string (or nil, if `omitempty` tag works for an inline table field). Let me know what you think @bitwiseguy